### ISSUE TITLE
Help Needed to Add Data to Firestore

### ISSUE THREAD

**Issue by just10minutes** [lupas/nuxt-fire#34]
@lupas Thank you very much for this package.

I have started using this, I did not get a example to achieve the below, your example mostly had adding record with ID, I am planning for auto populate ID.

I am planning to use Vuex to retrieve and add my data to FireStore.

Currently I have given read write public access to my firestore.

I do not have a fancy way of coding, I am still a new to all these, as in I do not have much experience. So my code below might look ugly. Apologies for that.

I need help in adding the data to my collection and get this data added back to my list of recipes.

What I have achieved till now -
I can access my data that is get list of recipes on my index page using below (Vuex code further below)
```
import { mapState, mapActions } from 'vuex'
computed: mapState({
    recipes: state => state.recipes.recipesList
  }),
  created() {
    this.fetchRecipes()
  },
  methods: {
    ...mapActions({
      fetchRecipes: 'recipes/fetchRecipes'
    })
  }
```
I am stuck at this stage now.
Page - AddRecipe
```
import { mapActions } from 'vuex'

methods: {
       ...mapActions({
      addRecipe: 'recipes/addRecipe'
    }),
    onSubmit: function(e) {   
      console.log(this.title)
      e.preventDefault()
      this.addRecipe(this.title)
    }
  }
```

I am getting the success alert and I can see data getting added but it is failing 

store/recipes.js

```
export const state = () => ({
  recipesList: []
})

export const getters = {
  // allRecipes: state => state.recipes
}

export const actions = {
  async fetchRecipes({ commit }) {
    const recipeArray = []
    const response = this.$fireStore.collection('recipes')
    try {
      await response
        .where('publish', '==', true)
        .orderBy('created', 'desc')
        .get()
        .then(querySnapshot => {
          querySnapshot.forEach(doc => {
            recipeArray.push(doc.data())
          })
        })
      commit('setRecipes', recipeArray)
    } catch (e) {
          // alert(e)
      alert('Error Fetching Data, please contact Webmaster')
    }
  },

  async addRecipe({ commit }, title) {
    const response = this.$fireStore.collection('recipes')

    try {
      await response
        .add({
          title: title,
          slug: title,
          publish: true,
          created: new Date()
        })
        .then(res => {
          // eslint-disable-next-line no-console
          console.log('Added document with ID: ', res.id) //This is printing data on console
           commit('addRecipe', res.data)
        })
    } catch (e) {
     
      console.log(e)

      alert(e)
      return
    }
    alert('Success.')
  }
}

export const mutations = {
  setRecipes: (state, recipesList) => (state.recipesList = recipesList),
  addRecipe: (state, recipe) => state.recipesList.unshift(recipe)
}


```

I hope you will be able to help me here. Thank you very much.

**Comment by lupas** [comment]
Hi @just10minutes, sorry for the late response, I am currently travelling.

The question seems to be rather a general Firestore question than anything related to this nuxt-fire module, but I can still try to help you. Though you might find an answer quicker on Stackoverflow.

You say you are getting the success alert and can see data getting added (I assume in the Firebase Web Console?) but you say it is failing.

What exactly is failing? What is the error message? If data gets added in Firestore you should be able to see the new data when you reload the page, right?

If you want to see the data added immediately in your vuex store, you might want to look into the .onSnapshot method to automatically update your store when data changes, see [here]( https://firebase.google.com/docs/firestore/query-data/listen).

You also might want to have a look at the [vuexfire module](https://github.com/vuejs/vuefire).

**Comment by just10minutes** [comment]
@lupas Thank you very much for getting back to me.

I didn't have any problem storing data to Firebase, Problem I faced was getting the result back. Then I realized I have to place another separate request to update my list. I was under impression it would happen at same time. I apologize.

Thank you for the feedback I will definitely get back to you if I face some issue. Since I am using nuxt fire do not want to use another package vuexfire, i will try to avoid more packages
